ppb/features/default_sprites.py

```diff
@@ -58,3 +60,118 @@ def _exponential_decay(self, time_delta):
+
+
+class KeyboardMovementSprite(ppb.Sprite):
+    """Sprite that moves with keyboard movement keys.
+
+    :param direction_keys: Configuration for mapping direction to keys
+    :param direction: Starting direction
+    :param acceleration: The increase in speed for a repeated key press in the following 0.3 seconds.
+    :param friction: The decrease in speed when there are no key presses in the following 0.3 seconds.
+    :param initial_speed: The speed per second when the first of subsequent keypresses are made.
+    :param speed: The current speed per second.
+    :param max_speed: Maximum speed per second that the sprite can travel in a specific direction.
+    :param min_speed: Minimum speed per second that the sprite can travel in a specific direction.
+    :param last_key_pressed_at: The datetime of the last time any key was pressed.
+    """
+    direction_keys: DirectionKeys = ArrowDirectionKeys()
+    direction: ppb.Vector = ppb.Vector(0, 0)
+    acceleration: float = 0.1
+    friction: float = 0.8
+    initial_speed: float = 1.0
+    speed: float = 0.0
+    max_speed: float = math.inf
+    min_speed: float = 0.0
+    last_key_pressed_at: datetime.datetime = datetime.datetime.utcnow()
+
+    def on_update(self, update_event, signal):
+        if self.max_speed < self.min_speed:
+            raise ValueError(f"{type(self).__name__} maximum speed cannot be less than minimum speed.")
+
+        if self.friction > 1:
+            raise ValueError(f"{type(self).__name__} friction cannot be more than 1.")
+
+        self._limit_speed()
+        self._stop_if_slow()
+
+        if not self.has_recent_keypress:
+            self.slow_down(update_event.time_delta)
+
+        self.position += self.direction * self.speed * update_event.time_delta
+
+
+    def on_key_pressed(self, key_event: ppb.events.KeyPressed, signal):
+        if self.has_recent_keypress:
+            self.accelerate(speed=self.speed)
+        else:
+            self.accelerate(speed=self.initial_speed)
+
+        if self.speed < self.initial_speed:
+            self.accelerate(speed=self.initial_speed)
+
+        if key_event.key == self.direction_keys.left:
+            self.direction += ppb.Vector(-1, 0)
+        elif key_event.key == self.direction_keys.right:
+            self.direction += ppb.Vector(1, 0)
+        elif key_event.key == self.direction_keys.up:
+            self.direction += ppb.Vector(0, 1)
+        elif key_event.key == self.direction_keys.down:
+            self.direction += ppb.Vector(0, -1)
+
+
+        self.last_key_pressed_at = datetime.datetime.utcnow()
+
+
+    def accelerate(self, speed: float):
+        self.speed = speed * (1 + self.acceleration)
+
+
+    def slow_down(self, time_delta: float):
+        self.speed = self.speed * (1 - self.friction) ** time_delta
+
+
+    @property
+    def has_recent_keypress(self) -> bool:
+        return datetime.datetime.utcnow() - self.last_key_pressed_at <= datetime.timedelta(microseconds=300000)
```

Comment:
This is not the preferred way to do clocks in PPB.

* It's not monotonic
* It'll do funky things if the scene is paused

The preferred way is to sum [`time_delta`](https://ppb.readthedocs.io/en/stable/reference/events.html#ppb.events.Update.time_delta) on the update event.

If that is not an option (eg, in the animation framework), there's `ppb.utils.get_time()`, but that has trade-offs (namely, it keeps running, even if the scene is not the active one).

Reply:
makes sense, will see what I can do